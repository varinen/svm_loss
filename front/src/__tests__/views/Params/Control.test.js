import React from "react";
import ParamsControl from "views/SVMLoss/Demo/Params/Control";
import {createShallow} from '@material-ui/core/test-utils';

describe('Params control renders correct grid', () => {
    let component, props, shallow;
    const mockUpdateParams = jest.fn();
    const mockFetchParams = jest.fn();
    beforeEach(() => {
        props = {
            params: {weights: [[1, 2], [3, 4]], biases: [1, 2]},
            updateParams: mockUpdateParams,
            fetchParams: mockFetchParams
        };
        shallow = createShallow({dive: true});
        component = shallow(<ParamsControl {...props}/>);
    });

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Generates a correct number of grid elements', () => {
        expect(component.find('GridItem').length).toEqual(6);
    });

    it('Change in a weight input calls a function to change a weight', () => {
        const w00 = 10;
        component.find('[label="w[0, 0]"]')
            .simulate('change', {target: {value: w00}});
        expect(mockUpdateParams).toHaveBeenCalledWith(0, 0, 'weights', w00);
    });

    it('Change in a bias input calls a function to change a bias', () => {
        const b0 = 10;
        component.find('[label="b[0]"]')
            .simulate('change', {target: {value: b0}});
        expect(mockUpdateParams).toHaveBeenCalledWith(0, 2, 'biases', b0);
    });

    it('Click on randomize button causes the fetchParams be called', () => {
        const b0 = 10;
        component.find('#rand-params')
            .simulate('click');
        expect(mockFetchParams).toHaveBeenCalledWith(1);
    });
});
