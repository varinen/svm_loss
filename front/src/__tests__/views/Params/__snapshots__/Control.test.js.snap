// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Params control renders correct grid renders correctly 1`] = `
<Fragment>
  <GridContainer
    className=""
    container={true}
    key="params-0"
    spacing={2}
  >
    <GridItem
      className=""
      key="w[0, 0]"
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        label="w[0, 0]"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={1}
      />
    </GridItem>
    <GridItem
      className=""
      key="w[0, 1]"
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        label="w[0, 1]"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={2}
      />
    </GridItem>
    <GridItem
      className=""
      key="b[0]"
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        label="b[0]"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={1}
      />
    </GridItem>
  </GridContainer>
  <GridContainer
    className=""
    container={true}
    key="params-1"
    spacing={2}
  >
    <GridItem
      className=""
      key="w[1, 0]"
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        label="w[1, 0]"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={3}
      />
    </GridItem>
    <GridItem
      className=""
      key="w[1, 1]"
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        label="w[1, 1]"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={4}
      />
    </GridItem>
    <GridItem
      className=""
      key="b[1]"
      xs={4}
    >
      <WithStyles(ForwardRef(TextField))
        label="b[1]"
        onChange={[Function]}
        step="0.1"
        type="number"
        value={2}
      />
    </GridItem>
  </GridContainer>
  <WithStyles(ForwardRef(Divider))
    className="ParamsControl-mt1-6 ParamsControl-mb1-10"
  />
  <ForwardRef
    color="primary"
    id="rand-params"
    onClick={[Function]}
  >
    Randomize
  </ForwardRef>
</Fragment>
`;
